activerecord/test/cases/adapters/mysql2/connection_test.rb

```diff
@@ -67,7 +67,7 @@ def test_execute_after_disconnect
   def test_quote_after_disconnect
     @connection.disconnect!
 
-    assert_raise(Mysql2::Error) do
+    assert_raise(ActiveRecord::StatementInvalid) do
```

Comment:
`StatementInvalid` seem like the wrong exception, but at least it's an `AcitveRecord::` one rather than leaking the adapter errors.